extensions/eclipse-che-theia-plugin-remote/src/node/hosted-plugin-remote.ts

```diff
@@ -160,7 +160,9 @@ export class HostedPluginRemote {
                 const entryName = getPluginId(deployedPlugin.metadata.model);
                 if (!this.hostedPluginMapping.getPluginsEndPoints().has(entryName)) {
                     this.hostedPluginMapping.getPluginsEndPoints().set(entryName, jsonMessage.endpointName);
-                    this.client.onDidDeploy();
+                    if (this.client) {
```

Comment:
I think it is better:
```
if (this.endpointsSockets.has(endpointAdress)) {
  return;
}
this.hostedPluginMapping.getPluginsEndPoints().set(entryName, jsonMessage.endpointName);
if (this.client === undefined) {
    return;
}
this.client.onDidDeploy();
```

Reply:
@olexii4 could you please elaborate on your suggested code?

Reply:
I just prefer ```if (this.client !== undefined) {``` instead of ```if(this.client) {``` as an alternative(more obvious as for me). 
Let it be as is.

Reply:
I am not sure that it is clearer...